src/rp2_common/pico_cyw43_arch/include/pico/cyw43_arch.h

```diff
@@ -281,6 +307,9 @@ static inline void cyw43_arch_lwip_end(void) {
  * single-core polling only (pico_cyw43_arch_poll) then there are no
  * locks to required, but it is still good practice to use this function.
  *
+ * \note as of SDK release 1.5.0, this is now equivalent to calling \ref async_context_lock_check
```

Comment:
Isn't the equivalent async_context_execute_sync not "check"

Reply:
oops, yes; i put this on the wrong function - i decided not to update lwip_protect as the async_context_execute_sync is NOT the exact equivalent